api/index.js

```diff
@@ -1,46 +1,56 @@
-const http = require('http');
-const cors = require('cors');
-const { OAuth2Strategy } = require('passport-google-oauth');
-const express = require('express');
-const session = require('express-session');
-const bodyParser = require('body-parser');
-const sessionFileStore = require('session-file-store');
-const passport = require('passport');
-const socketio = require('socket.io');
-const { oAuth: oAuthConfig, session: sessionConfig } = require('../config');
-const { initializeCache } = require('./data-access');
-const authRouter = require('./routers/auth');
-const albumsRouter = require('./routers/albums');
+const http = require("http");
+const cors = require("cors");
+const { OAuth2Strategy } = require("passport-google-oauth");
+const express = require("express");
+const session = require("express-session");
+const bodyParser = require("body-parser");
+const sessionFileStore = require("session-file-store");
+const passport = require("passport");
+const socketio = require("socket.io");
+const { oAuth: oAuthConfig, session: sessionConfig } = require("../config");
+const { initializeCache, joinPendingAlbums } = require("./albums/data-access");
+const authRouter = require("./user/auth");
+const albumsRouter = require("./albums/albums");
+const firebase = require("firebase");
+
+firebase.initializeApp(JSON.parse(process.env.FIREBASE_CONFIG));
 
 const app = express();
 const FileStore = sessionFileStore(session);
 
-app.use(cors({ origin: process.env.REACT_APP_BASE, credentials: true }));
-
 passport.serializeUser((user, done) => done(null, user));
 passport.deserializeUser((user, done) => done(null, user));
 passport.use(
   new OAuth2Strategy(
     oAuthConfig,
     // TODO: save the user to the database in this callback
-    (token, refreshToken, profile, done) => done(null, { profile, token })
+    async (token, refreshToken, profile, done) => {
+      console.log("token", token);
+      console.log("refreshToken", refreshToken);
+      console.log("profile", profile);
+
+      await joinPendingAlbums(token, profile.id);
```

Comment:
this is new

Reply:
Do we need to create the user in firebase here if they don't already exist?

Reply:
i think this is actually where we want to verify that the user is already in firebase from the admin right?

Reply:
Ah! Ok so create the user account in firebase when the admin adds them to the master list? That works 👍